src/main/java/by/custom/utilcalculator/service/CalculatorExceptM1.java

```diff
@@ -0,0 +1,155 @@
+package by.custom.utilcalculator.service;
+
+import by.custom.utilcalculator.domain.UserProgress;
+import by.custom.utilcalculator.domain.constants.Price;
+import by.custom.utilcalculator.domain.constants.steps.*;
+
+public class CalculatorExceptM1 {
+    private CalculatorExceptM1() {
+    }
+
+    private static class CalculatorExceptM1Holder {
+        private static final CalculatorExceptM1 CALCULATOR_INSTANCE = new CalculatorExceptM1();
+    }
+
+    public static CalculatorExceptM1 getInstance() {
+        return CalculatorExceptM1Holder.CALCULATOR_INSTANCE;
+    }
+
+    public String calculate(final UserProgress userProgress) {
```

Comment:
For me it looks like you can store the final price in the node as well.

Reply:
Sounds promising, but this is a magic for me. We should discuss ways how is it possible to implement this feature